scripts/consensus_merge.py

```diff
@@ -0,0 +1,636 @@
+    """
+    conflict_gt_tag = '0/1'
+    unan_tag, majo_tag, tie_tag = 'unanimous', 'majority', 'deadlock'
+
+    counts = Counter(gt_list)
+
+    if '.' in counts:
+        counts.pop('.')
+
+    if len(counts) == 1:
+        return list(counts)[0], unan_tag
+
+    largest_count = max(counts.values())
+    has_max_count = {k: v for k, v in counts.items() if v == largest_count}
+    if len(has_max_count) == 1:
+        return list(has_max_count)[0], majo_tag
+    else:
+        return conflict_gt_tag, tie_tag
+
+def get_ad_consensus(ad_list):
+    """ Get mean value of AD across callers 
+        Args:
+           ad_list (list): list of AD (comma-separated strings)
+        Return:
+           (tuple): mean depths
+    """
+    ref_depth = int(np.mean([int(f.split(',')[0]) for f in ad_list if f != '.']))
+    alt_depth = int(np.mean([int(f.split(',')[1]) for f in ad_list if f != '.']))
+    return (ref_depth, alt_depth)
+
+def get_af_consensus(af_list):
+    """ Get mean value of AF across callers
+        Args:
+            af_list (list): list of AF (strings)
+        Return:
+            (float): mean AF
+    """
+    return np.mean([float(f) for f in af_list if f != '.'])
+
+def get_dp_consensus(dp_list):
+    """ Get mean value of DP across callers
+        Args:
+            dp_list (list): list of DP (strings)
+        Return:
+            (int): mean DP
+    """
+    return int(np.mean([int(f) for f in dp_list if f != '.']))
+
+def get_mapq(cram, chrom, pos):
+    """ Get RMS mapping quality and number of MAPQ=0 reads at locus
+
+        Args:
+            cram (pysam.AlignmentFile): open in 'r' mode
+            chrom (str): locus chromosome name
+            pos (int): 1-based locus coordinate
+
+        Returns:
+            tuple of ints RMS mapq and # of mapq reads at locus
+    """
+    mapq = []
+    mq0 = 0
+#    print(chrom, pos-1, pos)
+
+    aligned_reads = cram.fetch(chrom, pos-1, pos)
+
+    for read in aligned_reads:
+        mapq.append(read.mapq)
+        if read.mapq == 0:
+            mq0 += 1
+
+    if not mapq:
+        return 0, 0
+
+    rms_mapq = np.sqrt(np.mean(np.array(mapq)**2))
+    return int(rms_mapq), mq0
+
+def build_output_record(single_caller_variants, output_vcf, normal_cram, hotspot=False):
+    """ Create a single VCF record for the consensus output
+            by interrogating the single-caller records for that variant
+            and add that record the consensus VCF
+
+        Args: 
+            single_caller_variants (list of Variants): list of Variants
+                representing a single call in one or more callers
+            output_vcf (pysam.VariantFile): consensus file, open in 'w' mode
+            normal_cram (pysam.Alignmentfile): Normal CRAM for this sample,
+                open in 'r' mode
+            hotspot (bool): whether this variant is in a mutational hotspot
+                (default False)
+    """
+    variant_lookup = {v.caller: v for v in single_caller_variants}
```

Comment:
So we talked about this but I think this should move to after the for loop at Line 492. It's not clear that for loop there actually updates the values in this dict and allows Lines 511-514 to work as written.

Reply:
It's in, thanks for pointing this out.